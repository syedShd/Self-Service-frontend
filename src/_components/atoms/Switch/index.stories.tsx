import { Story, Meta } from '@storybook/react'
import { ThemeProvider } from '@material-ui/core/styles'
import Switch from '.'
import baseTheme from '../../../core-utils/theme'

export default {
  title: 'atoms/Switch',
  component: Switch,
  argTypes: {
    color: {
      options: ['primary', 'secondary', 'default'],
      control: { type: 'select' },
      checked: {
        options: [true, false],
        control: { type: 'radio' },
      },
    },
  },
} as Meta

const Template: Story = ({ ...args }) => (
  <ThemeProvider theme={baseTheme}>
    <Switch {...args} />
  </ThemeProvider>
)

export const DefaultSwitch = Template.bind({})
DefaultSwitch.args = {
  checked: false,
}

export const PrimaryColorSwitch = Template.bind({})
PrimaryColorSwitch.args = {
  checked: true,
  color: 'primary',
}

export const DisabledSwitch = Template.bind({})
DisabledSwitch.args = {
  checked: true,
  disabled: true,
}
