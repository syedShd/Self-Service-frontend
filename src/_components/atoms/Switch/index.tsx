import * as React from 'react'
import { Switch as MUISwitch } from '@material-ui/core'
import PropTypes from 'prop-types'
import { getSwitchColors } from '../../../core-utils/helper'

type SwitchProps = {
  checked?: boolean
  setChecked?: (checked: boolean) => void
  color?: string
  ariaLabel?: string
  disabled?: boolean
  [restProps: string]: unknown
}

const Switch = ({
  checked,
  setChecked,
  color,
  ariaLabel,
  disabled,
  ...restProps
}: SwitchProps) => {
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setChecked?.(event.target.checked)
  }

  return (
    <MUISwitch
      checked={checked}
      onChange={handleChange}
      inputProps={{ 'aria-label': ariaLabel }}
      color={getSwitchColors(color)}
      disabled={disabled}
      {...restProps}
    />
  )
}

Switch.propTypes = {
  checked: PropTypes.bool,
  setChecked: PropTypes.func,
  color: PropTypes.string,
}

Switch.defaultProps = {
  checked: false,
  setChecked: null,
  color: 'primary',
  ariaLabel: 'Switch',
  disabled: false,
}

export default Switch
