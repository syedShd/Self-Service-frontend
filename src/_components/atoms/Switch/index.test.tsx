import React from 'react'
import { mount, shallow } from 'enzyme'
import { Switch as MUISwitch } from '@material-ui/core'
import Switch from '.'

describe('Switch Component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(<Switch />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Allows us to set props', () => {
    const wrapper = mount(<Switch checked ariaLabel="testLabel" />)
    expect(wrapper.props().checked).toEqual(true)
    expect(wrapper.props().ariaLabel).toEqual('testLabel')
    wrapper.setProps({ ariaLabel: 'test' })
    wrapper.setProps({ checked: false })
    expect(wrapper.props().ariaLabel).toEqual('test')
    expect(wrapper.props().checked).toEqual(false)
  })

  it('onChange works', () => {
    const mockHandleChangeText = jest.fn()
    const component = mount(
      <Switch
        name="field"
        setChecked={mockHandleChangeText}
        endAdorment="kg"
        margin="normal"
        variant="standard"
      />
    )
    component
      ?.find(MUISwitch)
      ?.props()
      ?.onChange?.(
        {
          target: { checked: true },
        } as React.ChangeEvent<HTMLInputElement>,
        true
      )
    expect(mockHandleChangeText).toHaveBeenCalledWith(true)
  })
})
