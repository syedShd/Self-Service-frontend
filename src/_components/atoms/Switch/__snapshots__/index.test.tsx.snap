// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component Renders correctly 1`] = `
<WithStyles(ForwardRef(Switch))
  checked={false}
  color="primary"
  disabled={false}
  inputProps={
    Object {
      "aria-label": "Switch",
    }
  }
  onChange={[Function]}
/>
`;
